cli/types/index.d.ts

```diff
@@ -811,14 +811,15 @@ declare namespace Cypress {
 
     /**
      * Invoke a function on the previously yielded subject.
-     * This isn't possible to strongly type without generic override yet.
-     * If called on an object you can do this instead: `.then(s => s.show())`.
-     * If called on an array you can do this instead: `.each(s => s.show())`.
-     * From there the subject will be properly typed.
      *
      * @see https://on.cypress.io/invoke
      */
-    invoke(functionName: keyof Subject, ...args: any[]): Chainable<Subject> // don't have a way to express return types yet
+    invoke<
+      TActualSubject extends (Subject extends Node[] ? JQuery<Subject> : (Subject extends Node ? JQuery<Subject> : Subject)),

```

Comment:
After reading through https://github.com/cypress-io/cypress/issues/806#issuecomment-339433409 and looking at the implementation of `invoke`, I added handling for the jQuery wrapping `invoke` does on its inputs